app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/image_import_workflow.rb

```diff
@@ -41,14 +41,28 @@ def post_execute
     cleanup_git_repository
 
     ems = ExtManagementSystem.find(options[:ems_id])
+    raise MiqException::Error, _("unable to find ems by this id '#{options[:ems_id]}'") if ems.nil?
```

Comment:
@iv1111 when you use `find(id)` if it cannot find the record it will raise a `RecordNotFound` exception and will never return `nil` so this `raise` is unreachable.

If you want to handle the not-found case yourself you can use `ems = ExtManagementSystem.find_by(:id => options[:ems_id])`  instead

Reply:
I'd need a custom handling code in order to form a user friendly message, so done as you suggested, using _find_by_.